flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigManager.java

```diff
@@ -0,0 +1,214 @@
+        return conf.clone();
+    }
+
+    private Configuration generateConfig(Tuple4<String, String, ObjectNode, Boolean> key) {
+        try {
+            LOG.info("Generating new {} config", key.f3 ? "deployment" : "observe");
+            return FlinkConfigBuilder.buildFrom(
+                    key.f0,
+                    key.f1,
+                    objectMapper.convertValue(key.f2, FlinkDeploymentSpec.class),
+                    defaultConfig,
+                    key.f3);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to load configuration", e);
+        }
+    }
+
+    private void scheduleConfigWatcher(KubernetesClient client) {
+        var checkInterval = defaultConfig.get(OPERATOR_DYNAMIC_CONFIG_CHECK_INTERVAL);
+        var millis = checkInterval.toMillis();
+        configUpdater = new ConfigUpdater(client);
+        Executors.newSingleThreadScheduledExecutor()
+                .scheduleAtFixedRate(configUpdater, millis, millis, TimeUnit.MILLISECONDS);
+        LOG.info("Enabled dynamic config updates, checking config changes every {}", checkInterval);
+    }
+
+    @VisibleForTesting
+    public void setWatchedNamespaces(Set<String> namespaces) {
+        this.namespaces = namespaces;
+        operatorConfiguration =
+                FlinkOperatorConfiguration.fromConfiguration(defaultConfig, namespaces);
+    }
+
+    @VisibleForTesting
+    public Runnable getConfigUpdater() {
+        return configUpdater;
+    }
+
+    private static Configuration readConfiguration(KubernetesClient client) {
+        var operatorNamespace = EnvUtils.getOrDefault(EnvUtils.ENV_OPERATOR_NAMESPACE, "default");
+        var configMap =
+                client.configMaps().inNamespace(operatorNamespace).withName(OP_CM_NAME).get();
+        Map<String, String> data = configMap.getData();
+        Path tmpDir = null;
+        try {
+            tmpDir = Files.createTempDirectory("flink-conf");
+            Files.write(
+                    tmpDir.resolve(GlobalConfiguration.FLINK_CONF_FILENAME),
+                    data.get(GlobalConfiguration.FLINK_CONF_FILENAME)
+                            .getBytes(Charset.defaultCharset()));
+            return GlobalConfiguration.loadConfiguration(tmpDir.toAbsolutePath().toString());
+        } catch (IOException ioe) {
+            throw new RuntimeException("Could not load default config", ioe);
+        } finally {
+            if (tmpDir != null) {
+                FileUtils.deleteDirectoryQuietly(tmpDir.toFile());
+            }
+        }
+    }
+
+    class ConfigUpdater implements Runnable {
+        KubernetesClient kubeClient;
+
+        public ConfigUpdater(KubernetesClient kubeClient) {
+            this.kubeClient = kubeClient;
+        }
+
+        public void run() {
+            try {
+                LOG.debug("Checking for config update changes...");
+                var newConf = readConfiguration(kubeClient);
```

Comment:
Why not read the config directly from the mounted dir. I think we do not have to interact with the configMap directly. Same at the the `FlinkConfigManager` construct

Reply:
Are the mounted files updated automatically if the configmap changes?

Reply:
As I know, it is

Reply:
nice, that will make this much simpler I agree

Reply:
I fixed this and verified that it works correctly